### PR TITLE
feat(core): run tests with debug asserts on

### PR DESCRIPTION
**Pull request by IceTDrinker**
### Resolves:

closes https://github.com/zama-ai/concrete-core-internal/issues/414

### Checklist 

(Use '[x]' to check the checkboxes, or submit the PR and then click the checkboxes)

* [ ] Tests for the changes have been added (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] The PR description links to the related issue (to link an issue, use '#XXX'.)
* [ ] The tests on AWS have been launched and are successful (comment with @slab-ci cpu_test and/or @slab-ci gpu_test to trigger the tests)
* [ ] The draft release description has been updated
* [ ] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

**Comment by IceTDrinker** [comment]
@slab-ci gpu_test

**Comment by IceTDrinker** [comment]
@slab-ci cpu_test

**Comment by IceTDrinker** [comment]
@slab-ci cpu_test

**Comment by IceTDrinker** [comment]
@slab-ci cpu_test

**Comment by IceTDrinker** [comment]
@slab-ci gpu_test

**Review by agnesLeroy — APPROVED** [review]

File: concrete-core-fixture/src/fixture/lwe_ciphertext_vector_glwe_ciphertext_discarding_private_functional_packing_keyswitch.rs [hunk 1]
```diff
@@ -290,7 +290,12 @@ where
         let raw_cleartext_vector =
             maker.transform_cleartext_vector_to_raw_vec(proto_cleartext_vector);
         let function_poly = Polynomial::from_container(raw_cleartext_vector);
-        let raw_input_vector = maker.transform_plaintext_vector_to_raw_vec(proto_plaintext_vector);
+        let mut raw_input_vector =
+            maker.transform_plaintext_vector_to_raw_vec(proto_plaintext_vector);
+        // raw_input_vector can be smaller than function_poly, currently fill_with_wrapping_mul
+        // requires the destination and source operands to have the same size, so we fill the first
+        // coefficiens with the raw_input_vector and the additional coefficients with 0
+        raw_input_vector.resize_with(function_poly.polynomial_size().0, Precision::Raw::zero);
```
Comment (agnesLeroy) [inline concrete-core-fixture/src/fixture/lwe_ciphertext_vector_glwe_ciphertext_discarding_private_functional_packing_keyswitch.rs:298]: We should probably add a check in the generic checks of the public API for this actually, no?

**Pull request merged**